### PR TITLE
Support for long double math.h functions

### PR DESCRIPTION
**Pull request by liammachado**
This pull request adds support for the long double versions of the math.h functions. It also adds regressions for these functions.

### CODE REVIEW

File: share/smack/lib/smack.c [hunk 1]
```diff
@@ -1090,61 +1197,80 @@ void __SMACK_decls(void) {
...
+    "$bitcast.bv80.bvlongdouble($load.bytes.bv80(M, p))}");
   D("function {:inline} $load.unsafe.bvhalf(M: [ref] i8, p: ref) returns (bvhalf) {"
     "$bitcast.i16.bvhalf($load.i16(M, p))}");
   D("function {:inline} $load.unsafe.bvfloat(M: [ref] i8, p: ref) returns (bvfloat) {"
     "$bitcast.i32.bvfloat($load.i32(M, p))}");
   D("function {:inline} $load.unsafe.bvdouble(M: [ref] i8, p: ref) returns (bvdouble) {"
     "$bitcast.i64.bvdouble($load.i64(M, p))}");
+  D("function {:inline} $load.unsafe.bvlongdouble(M: [ref] i8, p: ref) returns (bvlongdouble) {"
+    "$bitcast.i80.bvlongdouble($load.i80(M, p))}");
+  #endif
```
Comment (zvonimir) [inline share/smack/lib/smack.c]: Why did you add this `endif` here? Seems extraneous.

Comment (liammachado) [reply share/smack/lib/smack.c]: There were merge conflicts, and I accidentally kept this line. I've now removed it.

File: share/smack/lib/smack.c [hunk 2]
```diff
@@ -738,6 +795,30 @@ void __SMACK_decls(void) {
...
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv64, bvlongdouble, $ui2fp,(_ to_fp_unsigned 11 53));
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv56, bvlongdouble, $ui2fp,(_ to_fp_unsigned 11 53));
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv48, bvlongdouble, $ui2fp,(_ to_fp_unsigned 11 53));
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv40, bvlongdouble, $ui2fp,(_ to_fp_unsigned 11 53));
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv32, bvlongdouble, $ui2fp,(_ to_fp_unsigned 11 53));
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv24, bvlongdouble, $ui2fp,(_ to_fp_unsigned 11 53));
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv16, bvlongdouble, $ui2fp,(_ to_fp_unsigned 11 53));
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv8, bvlongdouble, $ui2fp,(_ to_fp_unsigned 11 53));
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv1, bvlongdouble, $ui2fp,(_ to_fp_unsigned 11 53));
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv128, bvlongdouble, $si2fp,(_ to_fp 11 53));
```
Comment (zvonimir) [inline share/smack/lib/smack.c]: Wait, these lines seem strange to me. Should this really be `_ to_fp 11 53`?
Doesn't `long double` have more bits typically?
